src/solvers/strings/string_refinement.cpp

```diff
@@ -645,7 +645,10 @@ decision_proceduret::resultt string_refinementt::dec_solve()
   for(equal_exprt &eq : equations)
   {
     if(can_cast_expr<function_application_exprt>(eq.rhs()))
+    {
+      simplify(eq.rhs(), ns);
```

Comment:
I was a bit confused by the distinction between `simplify` and `simplify_expr`; the second one should probably be called `simplify_copy`, but that's outside the scope of this PR, and they probably need some rework.